demonstrations/src/main/java/boofcv/demonstrations/segmentation/DemoSegmentColor.java

```diff
@@ -0,0 +1,277 @@
+		// change the size of the GUI to match the input image as well as handle zooming scale
+		BoofSwingUtil.invokeNowOrLater(() -> {
+			double zoom = BoofSwingUtil.selectZoomToShowAll(imageZoomPanel, width, height);
+			controls.setImageSize(width, height);
+			controls.setZoom(zoom);
+			imageZoomPanel.setScale(zoom);
+			imageZoomPanel.updateSize(width, height);
+			imageZoomPanel.getVerticalScrollBar().setValue(0);
+			imageZoomPanel.getHorizontalScrollBar().setValue(0);
+		});
+	}
+
+	// retrieve and process the user selected coordinates
+	public void clickProcessing( final BufferedImage image, int mouseX, int mouseY){
+
+		// accurate conversion of coordinates to image coordinates
+		Point2D_F64 imageCoordinates = imageZoomPanel.pixelToPoint(mouseX, mouseY);
+		double imageX = imageCoordinates.x;
+		double imageY = imageCoordinates.y;
+
+		// grab the rgb value at the image coordinates
+		int rgb = image.getRGB((int) imageX, (int) imageY);
+
+		//processes and populate color data array
+		processColor(rgb);
+	}
+
+	@Override
+	public void processImage( int sourceID, long frameID, BufferedImage buffered, ImageBase _input ) {
+
+		if(clickStatus == true){
+			clickProcessing(buffered, mouseX, mouseY);
+		}
+
+		float hue = color[0];
+		float saturation = color[1];
+
+		Planar<GrayF32> input = ConvertBufferedImage.convertFromPlanar(buffered, null, true, GrayF32.class);
+		Planar<GrayF32> hsv = input.createSameShape();
+
+		// Convert into HSV
+		ColorHsv.rgbToHsv(input, hsv);
+
+		// Euclidean distance squared threshold for deciding which pixels are members of the selected set
+		float maxDist2 = threshold * threshold;
+
+		// Extract hue and saturation bands which are independent of intensity
+		GrayF32 H = hsv.getBand(0);
+		GrayF32 S = hsv.getBand(1);
+
+		// Adjust the relative importance of Hue and Saturation.
+		// Hue has a range of 0 to 2*PI and Saturation from 0 to 1.
+		float adjustUnits = (float)(Math.PI/2.0);
+
+		// step through each pixel and mark how close it is to the selected color
+		var output = new BufferedImage(input.width, input.height, BufferedImage.TYPE_INT_RGB);
+
+		// start timer
+		long before = System.nanoTime();
+		for (int y = 0; y < hsv.height; y++) {
+			for (int x = 0; x < hsv.width; x++) {
+				// Hue is an angle in radians, so simple subtraction doesn't work
+				float dh = UtilAngle.dist(H.unsafe_get(x, y), hue);
+				float ds = (S.unsafe_get(x, y) - saturation)*adjustUnits;
+
+				// this distance measure is a bit naive, but good enough for to demonstrate the concept
+				float dist2 = dh*dh + ds*ds;
+				if (dist2 <= maxDist2) {
+					output.setRGB(x, y, buffered.getRGB(x, y));
+				}
+			}
+		}
+		// end timer
+		long after = System.nanoTime();
+		// calculate and update the processing time
+		controls.setProcessingTimeS((after - before)*1e-9);
+
+		// update the image panel
+		SwingUtilities.invokeLater(() -> {
+			if(clickStatus == true){
+				imageZoomPanel.setImage(output);
```

Comment:
Change `setImage()` to `setBufferedImageNoChange()` to stop it from changing the scale every time it repaints